src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,272 @@
+              color={isIncoming ? 'var(--positive-500)' : 'currentColor'}
+            >
+              {actionBalance}
+            </UIText>
+            <UIText kind="small/regular" color="var(--neutral-500)">
+              {actionValue}
+            </UIText>
+          </VStack>
+        </HStack>
+      </UnstyledButton>
+      <CenteredDialog
+        ref={dialogRef}
+        containerStyle={{ backgroundColor: 'var(--neutral-100)' }}
+        renderWhenOpen={() => (
+          <>
+            <Button
+              kind="ghost"
+              value="cancel"
+              size={36}
+              style={{
+                width: 36,
+                padding: 8,
+                position: 'absolute',
+                top: 16,
+                left: 8,
+              }}
+              onClick={handleDialogDismiss}
+            >
+              <ArrowLeftIcon style={{ width: 20, height: 20 }} />
+            </Button>
+            <ActionDetailedView
+              action={action}
+              networks={networks}
+              address={address}
+            />
+          </>
+        )}
+      />
+    </>
+  );
+}
+
+export function AssetHistory({
+  assetId,
+  address,
+  assetFullInfo,
+}: {
+  assetId: string;
+  address: string;
+  assetFullInfo?: AssetFullInfo;
+}) {
+  const { networks } = useNetworks();
+  const { singleAddressNormalized, ready } = useAddressParams();
+  const { currency } = useCurrency();
+  const {
+    value,
+    // TODO: this flag doesn't work, needs to be fixed
+    isFetching: actionsAreLoading,
+    hasNext,
+    fetchMore,
+  } = useAddressActions(
+    {
+      address: singleAddressNormalized,
+      currency,
+      actions_fungible_ids: [assetId],
+    },
+    {
+      limit: 10,
+      listenForUpdates: true,
+      paginatedCacheMode: 'first-page',
+      enabled: ready,
+    }
+  );
+
+  const asset = assetFullInfo?.fungible;
+
+  if (!asset || !networks || !value?.length) {
+    return null;
+  }
+
+  return (
+    <VStack gap={8} style={{ opacity: actionsAreLoading ? 0.8 : 1 }}>
+      <VStack gap={4}>
+        <UIText kind="headline/h3">History</UIText>
+        <VStack gap={0}>
+          {value.map((action) => (
+            <AssetHistoryItem
+              key={action.transaction.hash}
+              address={address}
+              networks={networks}
+              asset={asset}
+              action={action}
+            />
+          ))}
+        </VStack>
+      </VStack>
+      {hasNext ? (
+        <Button
+          kind="neutral"
+          onClick={fetchMore}
+          disabled={actionsAreLoading}
+          style={{
+            ['--button-background' as string]: 'var(--neutral-200)',
+            ['--button-background-hover' as string]: 'var(--neutral-300)',
+          }}
+        >
+          <HStack gap={8} alignItems="center" justifyContent="center">
+            <UIText kind="body/accent">More Transactions</UIText>
+            {actionsAreLoading ? <CircleSpinner /> : null}
```

Comment:
weird, I don't see this spinner (https://github.com/zeriontech/zerion-wallet-extension/pull/722/files#r2018965737)

Reply:
Yes, it looks like a problem with defi-sdk. Need to investigate.